engine/execution/computation/computer/computer.go

```diff
@@ -276,96 +288,111 @@ func (e *blockComputer) executeSystemCollection(
-	eventCounts, eventSize := res.ChunkEventCountsAndSize(collectionIndex)
-	e.metrics.ExecutionCollectionExecuted(time.Since(startedAt),
-		compUsed, memUsed,
-		eventCounts, eventSize,
-		snapshot.NumberOfRegistersTouched,
-		snapshot.NumberOfBytesWrittenToRegisters,
-		1,
-	)
-
 	return &flow.Collection{
-		Transactions: []*flow.TransactionBody{tx},
+		Transactions: collection.Transactions,
 	}, err
 }
 
 func (e *blockComputer) executeCollection(
 	blockSpan otelTrace.Span,
 	collectionIndex int,
 	txIndex uint32,
-	blockCtx fvm.Context,
 	collectionView state.View,
-	collection *entity.CompleteCollection,
+	collection collectionItem,
 	res *execution.ComputationResult,
 ) (uint32, error) {
 
-	e.log.Debug().
-		Hex("block_id", logging.Entity(blockCtx.BlockHeader)).
-		Hex("collection_id", logging.Entity(collection.Guarantee)).
-		Msg("executing collection")
-
 	// call tracing
 	startedAt := time.Now()
-	colSpan := e.tracer.StartSpanFromParent(blockSpan, trace.EXEComputeCollection)
-	defer func() {
+
+	var colSpan otelTrace.Span
+	if collection.isSystemCollection {
+		colSpan = e.tracer.StartSpanFromParent(
+			blockSpan,
+			trace.EXEComputeSystemCollection)
+
+		e.log.Debug().
+			Hex("block_id", logging.Entity(collection.ctx.BlockHeader)).
+			Msg("executing system collection")
+	} else {
+		colSpan = e.tracer.StartSpanFromParent(
+			blockSpan,
+			trace.EXEComputeCollection)
+
 		colSpan.SetAttributes(
 			attribute.Int("collection.txCount", len(collection.Transactions)),
-			attribute.String("collection.hash", collection.Guarantee.CollectionID.String()),
-		)
-		colSpan.End()
-	}()
+			attribute.String(
+				"collection.hash",
+				collection.Guarantee.CollectionID.String()))
+
+		e.log.Debug().
+			Hex("block_id", logging.Entity(collection.ctx.BlockHeader)).
+			Hex("collection_id", logging.Entity(collection.Guarantee)).
+			Msg("executing collection")
+	}
+
+	defer colSpan.End()
 
-	txCtx := fvm.NewContextFromParent(blockCtx, fvm.WithMetricsReporter(e.metrics), fvm.WithTracer(e.tracer))
 	for _, txBody := range collection.Transactions {
-		err := e.executeTransaction(txBody, colSpan, collectionView, txCtx, collectionIndex, txIndex, res, false)
+		err := e.executeTransaction(txBody, colSpan, collectionView, collection.ctx, collectionIndex, txIndex, res, false)
 		txIndex++
 		if err != nil {
 			return txIndex, err
 		}
 	}
+
+	if collection.isSystemCollection {
+		systemChunkTxResult := res.TransactionResults[len(res.TransactionResults)-1]
+		if systemChunkTxResult.ErrorMessage != "" {
+			// This log is used as the data source for an alert on grafana.
+			// The system_chunk_error field must not be changed without adding
+			// the corresponding changes in grafana.
+			// https://github.com/dapperlabs/flow-internal/issues/1546
+			e.log.Error().
+				Str("error_message", systemChunkTxResult.ErrorMessage).
+				Hex("block_id", logging.Entity(collection.ctx.BlockHeader)).
+				Bool("system_chunk_error", true).
```

Comment:
you can double-emit logs with both old and new names here so that we would be able to do the migration.

Reply:
add system_transaction double logging everywhere.  i'll look into renaming the variables later.